src/main/java/uk/gov/digital/ho/hocs/document/MSGDocumentConverter.java

```diff
@@ -0,0 +1,216 @@
+    }
+
+
+    private byte[] convertMsg(InputStream inputStream) throws IOException, DocumentException {
+
+        MsgParser parser = new MsgParser();
+        Message message = parser.parseMsg(inputStream);
+        MsgContents contents = extractContents(message);
+
+        Document pdf = new Document();
+        final ByteArrayOutputStream arrayOutputStream = new ByteArrayOutputStream();
+        PdfWriter.getInstance(pdf, arrayOutputStream);
+        pdf.open();
+        pdf.newPage();
+
+        pdf.add(new Paragraph(String.format("From: %s [%s]", contents.getFromEmail(),
+                                                 StringUtils.isEmpty(contents.getFromName()) ? "N/A" : contents.getFromName())));
+        pdf.add(new Paragraph(MessageFormat.format("To: {0}", StringUtils.isEmpty(contents.getToEmail()) ?
+                                                                   contents.getToName() : contents.getToEmail())));
+        pdf.add(new Paragraph("Subject: " + contents.getSubject()));
+
+        if (!parseElements(pdf, contents)) {
+            pdf.add(new Paragraph(""));
+            pdf.add(new Paragraph(contents.getBodyText()));
+        }
+        for (Attachment attachment : contents.getAttachments()) {
+            processAttachment(pdf, attachment);
+        }
+        pdf.close();
+
+        final byte[] outputBytes = arrayOutputStream.toByteArray();
+        arrayOutputStream.close();
+        return outputBytes;
+    }
+
+    private MsgContents extractContents(Message message) {
+        MsgContents contents = new MsgContents();
+        contents.setFromEmail(message.getFromEmail());
+        contents.setFromEmail(contents.getFromEmail().contains("Content_Types") ? StringUtils.EMPTY : contents.getFromName());
+        contents.setToEmail(message.getToEmail());
+        contents.setToName(message.getToName());
+        contents.setBodyText(message.getBodyText());
+        contents.setFromName(message.getFromName());
+        contents.setSubject(message.getSubject());
+        contents.setBodyHTML(message.getConvertedBodyHTML());
+        contents.setAttachments(message.getAttachments());
+        return contents;
+    }
+
+    private boolean parseElements(Document document, MsgContents contents) {
+        if (StringUtils.isEmpty(contents.getBodyHTML())) {
+            return false;
+        }
+        try {
+            ElementList elements = XMLWorkerHelper.parseToElementList(contents.getBodyHTML(), null);
+            document.add(new Paragraph(""));
+            for (Element element : elements) {
+                document.add(element);
+            }
+        } catch (RuntimeWorkerException | DocumentException | IOException e) {
+            return false;
+        }
+        return true;
+    }
+
+    private void processAttachment(Document pdf, Attachment attachment) {
+        // we don't process messages containing messages
+        if (!(attachment instanceof FileAttachment)) {
+            return;
```

Comment:
Scattered returns within methods are hard to follow, it would be nicer if we changed this if statement and included the logic for FileAttachment within it.

Reply:
Personal preference has no place in a review.